packages/py-api/py_api/controllers/hackathon/participants_controller.py

```diff
@@ -240,64 +241,70 @@ def add_participant(cls, participant: NewParticipant, jwt_token: str | None = No
-                            ),
-                        )
-                        TeamFunctionality.insert_team(newTeam)
-                        return JSONResponse(content=newTeam.model_dump(), status_code=200)
-                    else:
-                        return JSONResponse(content={"message": "Couldn't create team, because a team of the same name already exists!"}, status_code=422)
-
-                except (Exception) as e:
-                    return JSONResponse(content={"message": str(e)}, status_code=500)
+                return cls.add_participant_to_new_random_team(participant)
 
             else:
                 return JSONResponse(content={"message": "The maximum number of teams is reached."}, status_code=409)
+
+    @classmethod
+    def add_participant_to_existing_random_team(cls, available_team: HackathonTeam, participant: NewParticipant) -> JSONResponse:
+        try:
+            # Creates new participant
+            new_participant = ParticipantsFunctionality.create_participant(
+                participant,
+            )
+
+            new_participant_object_id = str(
+                new_participant.inserted_id,
+            )
+            # Adds the participant to the team
+            available_team = TeamFunctionality.add_participant_to_team_object(
+                available_team.team_name, new_participant_object_id,
+            )
+            cls.update_participant(
+                new_participant_object_id, UpdateParticipant(
+                    team_name=available_team.team_name,
+                ),
+            )
+            TeamFunctionality.update_team_query_using_dump(
+                available_team.model_dump(),
+            )
+            return JSONResponse(content=available_team.model_dump(), status_code=200)
+
+        except (Exception) as e:
+            return JSONResponse(content={"message": str(e)}, status_code=500)
+
+    @classmethod
+    def add_participant_to_new_random_team(cls, participant: NewParticipant) -> JSONResponse:
+        try:
+            # Creates new participant
+            new_participant = ParticipantsFunctionality.create_participant(
+                participant,
+            )
+
+            new_participant_object_id = str(
+                new_participant.inserted_id,
+            )
+            # Create New Team and assign the new participant to it
+            newTeam = TeamFunctionality.create_team_object_with_admin(
+                user_id=new_participant_object_id, team_name=TeamFunctionality.generate_random_team_name(), generate_random_team=True,
+            )
+
+            if newTeam:
+                cls.update_participant(
+                    new_participant_object_id, UpdateParticipant(
+                        team_name=newTeam.team_name,
+                    ),
+                )
+                TeamFunctionality.insert_team(newTeam)
+                return JSONResponse(content=newTeam.model_dump(), status_code=200)
```

Comment:
new line, or remove the else since if the if is not true the method returns otherwise. It just makes the code cleaner.